tests/suites/test_suite_nist_kw.function

```diff
@@ -229,75 +230,81 @@ void nist_kw_ciphertext_lengths(int in_len, int out_len, int mode, int res)
 exit:
     mbedtls_free(ciphertext);
     mbedtls_free(plaintext);
-    mbedtls_nist_kw_free(&ctx);
+    psa_destroy_key(key_id);
+    PSA_DONE();
 }
 /* END_CASE */
 
 /* BEGIN_CASE */
-void mbedtls_nist_kw_wrap(int cipher_id, int mode, data_t *key, data_t *msg,
-                          data_t *expected_result)
-{
-    unsigned char result[528];
-    mbedtls_nist_kw_context ctx;
-    size_t result_len, i, padlen;
+void mbedtls_nist_kw_wrap(int mode, const data_t *key, data_t *msg,
+                          data_t *expected_result ) {
+    mbedtls_svc_key_id_t key_id = MBEDTLS_SVC_KEY_ID_INIT;
+    size_t result_length, padlen;
 
-    mbedtls_nist_kw_init(&ctx);
+    padlen = (msg->len % 8 != 0) ? 8 - (msg->len % 8) : 0;
+    int result_size = msg->len + 8 + padlen;
+    unsigned char *result;
+    result = mbedtls_calloc(1, result_size);
 
-    memset(result, '+', sizeof(result));
+    psa_key_attributes_t attributes = PSA_KEY_ATTRIBUTES_INIT;
 
-    TEST_ASSERT(mbedtls_nist_kw_setkey(&ctx, cipher_id,
-                                       key->x, key->len * 8, 1) == 0);
+    PSA_INIT();
 
-    /* Test with input == output */
-    TEST_ASSERT(mbedtls_nist_kw_wrap(&ctx, mode, msg->x, msg->len,
-                                     result, &result_len, sizeof(result)) == 0);
+    psa_set_key_usage_flags(&attributes, PSA_KEY_USAGE_ENCRYPT |
```

Comment:
Please only set `ENCRYPT` here, for the wrap test.

I understand that for tests that call both wrap and unwrap it's convenient to set both, so that we don't have to have twice as many keys. But here we're only doing wrap, so we should really test that it works with only ENCRYPT set, as the documentation says.